workflow-bot-app/src/main/java/com/symphony/bdk/workflow/engine/WorkflowDirectGraph.java

```diff
@@ -0,0 +1,121 @@
+package com.symphony.bdk.workflow.engine;
+
+import com.symphony.bdk.workflow.swadl.v1.Workflow;
+
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * A direct graph class representing the SWADL workflow
+ *
+ * @see Workflow
+ */
+public class WorkflowDirectGraph {
```

Comment:
My concern here is about memory overflow in case of a complex workflow with small memory allocated.